Doc/reference/datamodel.rst

```diff
@@ -103,15 +103,18 @@ of the immediately contained objects are implied.  So, if an immutable container
 (like a tuple) contains a reference to a mutable object, its value changes if
 that mutable object is changed.
 
-Types affect almost all aspects of object behavior.  Even the importance of
-object identity is affected in some sense: for immutable types, operations that
-compute new values may actually return a reference to any existing object with
-the same type and value, while for mutable objects this is not allowed.  E.g.,
-after ``a = 1; b = 1``, ``a`` and ``b`` may or may not refer to the same object
-with the value one, depending on the implementation, but after ``c = []; d =
-[]``, ``c`` and ``d`` are guaranteed to refer to two different, unique, newly
-created empty lists. (Note that ``c = d = []`` assigns the same object to both
-``c`` and ``d``.)
+Types affect almost all aspects of object behavior. Even the importance of
+object identity is affected in some sense.
+
+For immutable types such as :class:`int` or :class:`str`, operations that
+compute new values may actually return a reference to any existing object
+with the same type and value, e.g., after ``a = 1; b = 1``, *a* and *b* may
+or may not refer to the same object with the value one.
+
+For mutable types such as :class:`list` or :class:`dict`, this is not allowed,
+e.g., after ``c = []; d = []``, *c* and *d* are guaranteed to refer to two
+different, unique, newly created empty lists (note that ``e = f = []`` assigns
+the *same* object to both *e* and *f*).
```

Comment:
I think the current wording reads better in general, but it helps to add some clarifying points. We don't need three paragraphs here, so I've recombined.

```suggestion
Types affect almost all aspects of object behavior.  Even the importance of
object identity is affected in some sense: for immutable types, operations that
compute new values may actually return a reference to any existing object with
the same type and value, while for mutable objects this is not allowed.
For example, after ``a = 1; b = 1``, ``a`` and ``b`` may or may not refer to 
the same objectw ith the value one, depending on the implementation.
This is because :class:`int` is an immutable type, so the reference to ``1``
can be reused. This behaviour is dependant on the implementation used,
so should not be relied upon, but is something to be aware of when making use of
object identity tests.
However, after ``c = []; d = []``, *c* and *d* are guaranteed to refer to two
different, unique, newly created empty lists. (Note that ``e = f = []`` assigns
the *same* object to both *e* and *f*).
```

Reply:
Yeah, 3 paragraphs are actually too separated once they are in an HTML format. From an RST PoV, it's good but when it's rendered it's not good.